ls: silently ignore `-T` option

Stonks3141 opened the pull request:
The option can't be implemented without a significant rewrite according to #3624, and right now passing the option causes a failure.

* fixed error when `-T` option is passed
* write tests for `-T` option

sylvestre commented:
i don't think we can mix clap 3.1 & 3.2
see
https://github.com/uutils/coreutils/issues/3634#issuecomment-1154870604=

Stonks3141 commented:
Ok, I reverted the changes and removed the relevant tests.

sylvestre commented:
many CI jobs are failing, could you please have a look?
thanks

Stonks3141 commented:
They all say
```text
error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.64.0-nightly (23e21bdd2 2022-07-15) running on x86_64-unknown-linux-gnu

note: compiler flags: --crate-type lib -C embed-bitcode=no -C debuginfo=2 -C incremental -Z binary-dep-depinfo -Z save-analysis
```
I don't think I changed anything that would break CI, I just added a Clap arg and a test function. I can replicate the error locally, but the error is in `uu_od`. It also occurs if I clone the upstream repo and run `cargo +nightly udeps` (udeps needs nightly).

Stonks3141 commented:
Most of the CI jobs magically fixed themselves, and I fixed the style/spelling issues. Should be ready to merge now, unless you have any other issues?

sylvestre commented on tests/by-util/test_ls.rs:
```diff
@@ -3117,3 +3117,13 @@ fn test_dereference_symlink_file_color() {
+    let scene = TestScenario::new(util_name!());
+
+    scene.ucmd().args(&["-T", "3"]).succeeds();
+
```
please run rustfmt on this file too

Stonks3141 replied:
Should be done, there was a trailing newline.

sylvestre commented on tests/by-util/test_ls.rs:
```diff
@@ -3117,3 +3117,12 @@ fn test_dereference_symlink_file_color() {
+fn test_tabsize_option() {
+    let scene = TestScenario::new(util_name!());
+
+    scene.ucmd().args(&["-T", "3"]).succeeds();
```
I would also like to see a new test with what it should actually return once it is implemented
and marked as ignored.
Could you please do that ? thanks

Stonks3141 replied:
Ok, done.

Stonks3141 replied:
Spellcheck doesn’t like “aaaaaaaa”, “bbbb”, etc. Should I just add them to the spell ignore?

sylvestre replied:
yeah
